Client/mods/deathmatch/logic/luadefs/CLuaVectorGraphicDefs.cpp

```diff
@@ -0,0 +1,126 @@
+        CLuaCFunctions::AddFunction(name, func);
+}
+
+void CLuaVectorGraphicDefs::AddClass(lua_State* luaVM)
+{
+    lua_newclass(luaVM);
+
+    lua_classfunction(luaVM, "create", "svgCreate");
+    lua_classfunction(luaVM, "getDocumentXML", "svgGetDocumentXML");
+    lua_classfunction(luaVM, "SetDocumentXML", "svgSetDocumentXML");
+
+    lua_registerclass(luaVM, "SVG");
+}
+
+CClientVectorGraphic* CLuaVectorGraphicDefs::SVGCreate(lua_State* luaVM, CVector2D size, std::optional<std::string> pathOrRawData)
+{
+    if (size.fX <= 0 || size.fY <= 0)
+    {
+        throw std::invalid_argument("A vector graphic must be atleast 1x1 in size.");
+    }
+
+    CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
+    CResource* pParentResource = pLuaMain->GetResource();
+
+    CClientVectorGraphic* pVectorGraphic = g_pClientGame->GetManager()->GetRenderElementManager()->CreateVectorGraphic((int)size.fX, (int)size.fY);
+
+    if (pVectorGraphic)
+    {
+        // Make it a child of the resource's file root ** CHECK  Should parent be pFileResource, and element added to pParentResource's ElementGroup? **
+        pVectorGraphic->SetParent(pParentResource->GetResourceDynamicEntity());
+
+        // Set our owner resource
+        pVectorGraphic->SetResource(pParentResource);
+
+        if (pathOrRawData.has_value())
+        {
+            std::string strRawData = pathOrRawData.value().c_str();
+            std::size_t szCharPos = strRawData.find_first_not_of(" \t", 0);
+
+            const char charAuto = strRawData.at(szCharPos);
+
+            if (charAuto == '<')
+            {
+                if (!pVectorGraphic->LoadFromData(pathOrRawData.value()))
+                {
+                    pVectorGraphic->Destroy();
+                    throw std::invalid_argument("Unable to load raw SVG data (check for syntax errors)");
+                }
+            }
+            else
+            {
+                std::string strPath;
+                if (CResourceManager::ParseResourcePathInput(pathOrRawData.value(), pParentResource, &strPath) && FileExists(strPath))
+                {
+                    if (!pVectorGraphic->LoadFromFile(strPath))
+                    {
+                        pVectorGraphic->Destroy();
+                        throw std::invalid_argument(SString("Unable to load SVG (check for syntax errors) [%s]", pathOrRawData.value().c_str()));
+                    }
+                }
+                else
+                {
+                    throw std::invalid_argument(SString("Unable to load SVG (file doesn't exist) [%s]", pathOrRawData.value().c_str()));
+                }
+            }
+        }
+
+        return pVectorGraphic;
+    }
+
+    throw std::invalid_argument("Unknown error occurred creating SVG element");
+}
+
+CXMLNode* CLuaVectorGraphicDefs::SVGGetDocumentXML(lua_State* luaVM, CClientVectorGraphic* pVectorGraphic)
+{
+    CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(luaVM);
```

Comment:
Maybe we should add CLuaMain* to the argument parser to avoid this recurring pattern

Reply:
Agreed, we should open another issue for that.